docs/advanced-features/good-practice.md

```diff
@@ -101,5 +103,789 @@ If enabled (see set-up instructions above), these events will be
+
+User is granted with admin rights.
+
+- `requestParameters`
+  - `username: str`
+
+###### `Users.RevokeAdmin`
+
+User is revoked admin rights.
+
+- `requestParameters`
+  - `username: str`
+
+###### `Users.Delete`
+
+User deleted.
+
+- `requestParameters`
+  - `username: str`
+
+###### `Users.ResetPassword`
+
+User's password reset.
+
+- `requestParameters`
+  - `username: str`
+
+###### `Users.SetRole`
+
+User's role updated.
+
+- `requestParameters`
+  - `username: str`
+  - `role: str` Role name
+
+##### `Buckets` namespace
+
+Bucket management operations.
+Only accessible by the admin users.
+
+###### `Buckets.Add` (GraphQL: `Mutation.bucketAdd`)
+
+Bucket added to the stack.
+
+###### `Buckets.Update` (GraphQL: `Mutation.bucketUpdate`)
+
+Bucket settings updated.
+
+###### `Buckets.Remove` (GraphQL: `Mutation.bucketRemove`)
+
+Bucket removed from the stack.
+
+##### `Policies` namespace
+
+Quilt Policy management operations.
+Only accessible by the admin users.
+
+###### `Policies.Create` (GraphQL: `Mutation.policyCreateManaged` / `Mutation.policyCreateUnmanaged`)
+
+Quilt Policy created.
+
+`requestParameters.managed` is set accordingly.
+
+###### `Policies.Update` (GraphQL: `Mutation.policyUpdateManaged` / `Mutation.policyUpdateUnmanaged`)
+
+Quilt Policy updated.
+
+`requestParameters.managed` is set accordingly.
+
+###### `Policies.Delete` (GraphQL: `Mutation.policyDelete`)
+
+Quilt Policy deleted.
+
+##### `Roles` namespace
+
+Quilt Role management operations.
+Only accessible by the admin users.
+
+###### `Roles.Create` (GraphQL: `Mutation.roleCreateManaged` / `Mutation.roleCreateUnmanaged`)
+
+Quilt Role created.
+
+`requestParameters.managed` is set accordingly.
+
+###### `Roles.Update` (GraphQL: `Mutation.roleUpdateManaged` / `Mutation.roleUpdateUnmanaged`)
+
+Quilt Role updated.
+
+`requestParameters.managed` is set accordingly.
+
+###### `Roles.Delete` (GraphQL: `Mutation.roleDelete`)
+
+Quilt Role deleted.
+
+###### `Roles.SetDefault` (GraphQL: `Mutation.roleSetDefault`)
+
+Quilt Role set as default.
+
+##### `Scripts` namespace
+
+Admin scripts. Usually invoked by CloudFormation on stack bring-up / upgrade.
+
+All `Scripts.*` events contain the following data:
+
+- `additionalRequestData`
+  - `script_name: str` Script filename
+  - `script_args: str[]` List of arguments
+  - `script_command: str` The whole unparsed command string
+
+- `userIdentity: IAMUser` AWS identity of the user executing the script
+
+- `requestParameters: object` Parsed named script arguments
+
+###### `Scripts.CreateAdmin`
+
+Create an admin user account.
+Succeeds once on stack bring-up, fails on subsequent stack upgrades.
+
+- `requestParameters`
+  - `env: bool` Pass account info in environment variables
+  - `role_name: str` (optional) Name for Quilt T4 Role
+  - `email: str`
+  - `password: "***"` (optional)
+
+- `additionalEventData`
+  - `role_id: str` (optional)
+    ID of the Quilt Role, if found by the name `role_name` (when provided).
```

Comment:
IMO item description should be visually separated, so I'd do either
```suggestion
  - `role_id: str` (optional):
    ID of the Quilt Role, if found by the name `role_name` (when provided).
```

or 

```suggestion
  - `role_id: str` (optional)
  
    ID of the Quilt Role, if found by the name `role_name` (when provided).
```

and use for every list one of these styles.